libcommon/include/pivot/pivot.hxx

```diff
@@ -5,23 +5,32 @@
 #include <cpplogger/Logger.hpp>
 #include <cpplogger/utils/source_location.hpp>
 
+#include "pivot/Platform.hxx"
+#include "pivot/debug.hxx"
+
 #ifndef NDEBUG
     #define DEBUG_FUNCTION logger.trace(::function_name()) << "Entered";
 
-    #define pivot_assert(expr, msg) (static_cast<bool>(expr) ? void(0) : __pivot_assert_failed(#expr, msg))
-
-    #define __pivot_assert_failed(expr, msg)   \
-        (/* detect catch2 at runtime */ false) \
-            ? (void(0))                        \
-            : (logger.err(::file_position()) << "Assertion failed: " #expr " :: " << msg, logger.stop(), std::abort())
+    #define pivot_assert(expr, msg)                                              \
+        {                                                                        \
+            if (UNLIKELY(!(expr))) {                                             \
+                __pivot_assert_failed(#expr, msg);                               \
+                if (pivot::plateform::isDebuggerPresent()) { PLATFORM_BREAK(); } \
+                std::abort();                                                    \
+            }                                                                    \
+        }
 
-    #define pivot_check(expr, msg) (static_cast<bool>(expr) ? void(0) : __pivot_check_failed(#expr, msg))
+    #define __pivot_assert_failed(expr, msg)                                       \
+        logger.err(::file_position()) << "Assertion failed: " #expr " :: " << msg; \
+        logger.stop();
 
-    #define __pivot_check_failed(expr, msg) (logger.warn(::file_position()) << #expr ": " << msg, void(0))
+    #define pivot_check(expr, msg) \
+        ((LIKELY(!!(expr))) || __pivot_check_failed(#expr, msg)) || ([]() { if (pivot::plateform::isDebuggerPresent()) { PLATFORM_BREAK(); }}(), false)
```

Comment:
Bad idea to put likely here. That means if ever we have a check which fails in a tight loop, the performance will be disastrous, and the program unusable. The optimizer knows what it's doing, let it do the work of figuring which branch is more likely.

Reply:
If we have a check that fail in a loop, there is a bigger problem than performance. Plus I prefer loosing some time in a error scenario than losing time in a success scenario.

Reply:
Plus, "/proc/self/status" is in RAM, so the function is really cheap.

Reply:
You will not lose time in either scenario if you remove the likely. Of course, we don't actually care about the performance, but as a model for other project which would actually matter, I would refuse to accept such changes without benchmarks, as it clearly seems to be premature optimization.